node/feature_flags.ts

```diff
@@ -1,9 +1,10 @@
-const defaultFlags: Record<string, boolean> = {
+const defaultFlags = {
   edge_functions_fail_unsupported_regex: false,
+  edge_functions_invalid_config_throw: false,
 }
 
 type FeatureFlag = keyof typeof defaultFlags
-type FeatureFlags = Record<FeatureFlag, boolean>
+type FeatureFlags = Record<FeatureFlag | string, boolean>
```

Comment:
Can you explain this change?

Reply:
`FeatureFlag` was of type `string`, because of line 1. I remove the type on `defaultFlags` and because of that the type of `FeatureFlag ` is now `"edge_functions_fail_unsupported_regex"|"edge_functions_invalid_config_throw".

So the the final type for `FeatureFlags` is now Record<"edge_functions_fail_unsupported_regex"|"edge_functions_invalid_config_throw" | string, boolean> which provides autocompletion for the default flags, which wasn't the case before.

Reply:
I'm asking because the idea of the `FeatureFlags` type was that we shouldn't use any flag in the codebase that is not in the `defaultFlags` object (maybe the name `defaultFlags` is misleading).

Reply:
exactly, now it provides the `defaultFlags` as dropdown, but still allows string.

but i could change it to

```suggestion
type FeatureFlag = keyof typeof defaultFlags | string
type FeatureFlags = Record<FeatureFlag, boolean>
```

Then `FeatureFlag` (singular) also allows any FF. WDYT?

Reply:
> Oh misread your comment. I can remove the `|string` again, then only the flags in `defaultFlags` ar valid.

Yeah, I think that's what we want. Unless you think of a reason to allow arbitrary strings?

Reply:
Ok, I used partial now, because in some places we feed an empty object, so the flags might be boolean or undefined, but only the flags in `defaultFlags` are allowed to be used:

<img width="887" alt="Screenshot 2023-03-07 at 13 24 40" src="https://user-images.githubusercontent.com/231804/223421601-857d44f1-23f4-4671-a8ac-11816e809063.png">

<img width="975" alt="Screenshot 2023-03-07 at 13 24 16" src="https://user-images.githubusercontent.com/231804/223421613-cf1081da-976e-4c39-90df-99316993f92e.png">